core/shared/src/main/scala/cats/parse/RadixNode.scala

```diff
@@ -81,24 +86,21 @@ object RadixNode {
       case None =>
         leaf
     }
-  }
 
   private val leaf = new RadixNode(Array.empty, Array.empty, Array.empty, true)
 
-  private def commonPrefix(s1: String, s2: String): Int = {
-    @tailrec
-    def loop(idx: Int): Int =
-      if (idx >= s1.size || idx >= s2.size) {
-        idx
+  final def commonPrefixLength(s1: String, s2: String): Int = {
+    val len = Integer.min(s1.length, s2.length)
```

Comment:
minor optimization to do only one comparison below, and do it against a local not the result of a method call.